app/controllers/login_controller.rb

```diff
@@ -0,0 +1,13 @@
+class LoginController < ApplicationController
+
+  def create
+    @market = Market.find(params[:id])
+    redirect_to market_path(@market.id)
+  end
+
+  def vendorcreate
+    @vendor = Vendor.find(params[:id])
+    redirect_to vendor_path(@vendor.id)
+  end
+
```

Comment:
@market and @vendor should be local variables here. Instance variables should be used for elements that you need to access in the view, and since you are doing a redirect here you will hit the route and controller that you specified and generate any needed instance variables there.